sdk/compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs

```diff
@@ -1026,10 +1053,14 @@ convertExports env mc = do
             flSelector <- convertQualName (flSelector f)
             pure f { flSelector }
 
-        mkExportDef :: Integer -> ExportInfo -> Definition
-        mkExportDef i info =
+        mkExportDef :: LF.ExprValName -> ExportInfo -> Definition
+        mkExportDef name info =
             let exportType = encodeExportInfo info
-            in DValue (mkMetadataStub (exportName i) exportType)
+            in DValue (mkMetadataStub name exportType)
+
+        -- Tag for explicit exports, so we can differentiate between no exports and old-style exports
```

Comment:
"old-style" will be a bit obscure when someone reads that in a few years, maybe something like:

>  "export everything", as we did in old dars before the introduction of $$explicitExport